src/fips-utils/src/main/java/org/geoserver/security/MigrateKeystore.java

```diff
@@ -0,0 +1,249 @@
+package org.geoserver.security;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.security.KeyStore;
+import java.security.Security;
+import java.util.Enumeration;
+import org.apache.commons.cli.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Standalone keystore migration utility for FIPS compliance. This utility can migrate keystores between different
+ * formats and providers.
+ */
+public class MigrateKeystore {
+
+    private static final Logger logger = LoggerFactory.getLogger(MigrateKeystore.class);
+
+    // Static initialization to register BouncyCastle provider
+    static {
+        try {
+            // Register BouncyCastle provider if not already registered
+            if (Security.getProvider("BC") == null) {
+                Security.addProvider(new org.bouncycastle.jce.provider.BouncyCastleProvider());
+                logger.info("BouncyCastle provider registered successfully");
+            }
+        } catch (Exception e) {
+            logger.warn("Could not register BouncyCastle provider: " + e.getMessage());
+        }
+    }
+
+    /** Main method for keystore migration */
+    public static void main(String[] args) {
+        Options options = new Options();
+
+        options.addOption("s", "source", true, "Source keystore file (required)");
+        options.addOption("sp", "source-password", true, "Source keystore password (required)");
+        options.addOption("t", "target", true, "Target keystore file (required)");
+        options.addOption("tp", "target-password", true, "Target keystore password (required)");
+        options.addOption("st", "source-type", true, "Source keystore type (default: auto-detect)");
+        options.addOption("spr", "source-provider", true, "Source keystore provider (default: auto-detect)");
+        options.addOption("tt", "target-type", true, "Target keystore type (default: BCFKS)");
+        options.addOption("tpr", "target-provider", true, "Target keystore provider (default: auto-detect)");
+        options.addOption("v", "verbose", false, "Verbose output");
+        options.addOption("h", "help", false, "Show help");
```

Comment:
looks like this was duplicated somehow. The FIPSKeystoreProvider also has this same main.